### PR TITLE
Disable Travis CI deployment using SHOULD_DEPLOY flag.

### PR DESCRIPTION
**Pull request by ewelinajozwik**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -21,13 +21,15 @@ env:
...
      - GITHUB_USERNAME="fairspace-ci"
      - GCP_PROJECT="fairspace-207108"
-     - SNAPSHOT_CONTEXT="gke_fairspace-207108_europe-west1-b_fairspacecicluster"
-     - SNAPSHOT_CLUSTER_NAME="fairspacecicluster"
-     - SNAPSHOT_CLUSTER_ZONE="europe-west1-b"
-     - RELEASE_CONTEXT="gke_fairspace-207108_europe-west1-b_fairspacecicluster"
-     - RELEASE_CLUSTER_NAME="fairspacecicluster"
-     - RELEASE_CLUSTER_ZONE="europe-west1-b"
      - GIT_TAG_FILES_OVERRIDE="charts/fairspace/Chart.yaml charts/fairspace/values.yaml projects/"
+     - SHOULD_DEPLOY=0
```
Comment (gijskant) [inline .travis.yml]: Maybe use 'false' for better readability?
```suggestion
     - SHOULD_DEPLOY=false
```

File: .travis.yml [hunk 2]
```diff
@@ -47,10 +49,10 @@ before_install:
   - export KUBE_CONFIG_ENC_IV=$encrypted_5162cfbd2a53_iv
   - git clone https://$GITHUB_USERNAME:$GITHUB_PASSWORD@github.com/$BUILD_SCRIPTS_REPO ci
   - export BUILD_SCRIPTS_DIR=$(pwd)/ci
-  - git clone https://$GITHUB_USERNAME:$GITHUB_PASSWORD@github.com/$DEPLOYMENT_CONFIG_REPO ci-config
-  - export DEPLOYMENT_CONFIG_DIR="$(pwd)/ci-config/fairspace"
-  - 'if [[ "$TRAVIS_BRANCH" = "$SNAPSHOT_BRANCH" ]]; then export ALLOW_SNAPSHOTS=1 DEPLOY_TARGET=ci DEPLOY_PLATFORM="GCP" KUBERNETES_CONTEXT="$SNAPSHOT_CONTEXT" CLUSTER_NAME="$SNAPSHOT_CLUSTER_NAME" CLUSTER_ZONE="$SNAPSHOT_CLUSTER_ZONE" ; fi'
-  - 'if [[ "$TRAVIS_BRANCH" = "$RELEASE_BRANCH" ]]; then export DEPLOY_TARGET=ci DEPLOY_PLATFORM="GCP" KUBERNETES_CONTEXT="$RELEASE_CONTEXT" CLUSTER_NAME="$RELEASE_CLUSTER_NAME" CLUSTER_ZONE="$RELEASE_CLUSTER_ZONE" ; fi'
+  - 'if [[ "$SHOULD_DEPLOY" ]]; then git clone https://$GITHUB_USERNAME:$GITHUB_PASSWORD@github.com/$DEPLOYMENT_CONFIG_REPO ci-config ; fi'
```
Comment (gijskant) [inline .travis.yml]: I don't think this works (at least not in bash). Maybe `if [[ "$SHOULD_DEPLOY" = "true" ]]; then`?

File: .travis.yml [hunk 3]
```diff
@@ -81,13 +83,13 @@ jobs:
...
-#     - stage: deploy
-#       cache: false
-#       if: branch = dev and type != pull_request
-#       install:
-#         - source charts/.travis/install.sh
-#       script:
-#       - travis_wait ci/helm/deploy.sh $DEPLOY_TARGET -f projects/saturn/views.yaml
+     - stage: deploy
+       cache: false
+       if: $SHOULD_DEPLOY and branch = dev and type != pull_request
```
Comment (gijskant) [inline .travis.yml]: You probably need the function `env` here (see https://docs.travis-ci.com/user/conditions-v1):
```suggestion
       if: env(SHOULD_DEPLOY) = true and branch = dev and type != pull_request
```
Although I remember that there were issues with using environment variables in conditions in Travis scripts.

File: .travis.yml (hunk 1 above)
Comment (ewelinajozwik) [reply .travis.yml]: I wanted to be consistent with `SHOULD_RELEASE` flag from [fairspace-build-scripts repo](https://github.com/thehyve/fairspace-build-scripts/blob/master/setup_env.sh#L13), but it looks like that one is either `1` or `undefined`. I'll make the deploy one `true` \ `false` then.

**Comment by ewelinajozwik** [comment]
There is still something wrong with this change, travis build is not even triggered for this PR.

File: .travis.yml (hunk 1 above)
Comment (gijskant) [reply .travis.yml]: Weird language, this 'bash'... :)

Comment (gijskant) [reply .travis.yml]: So, `[[ "$CONDITION" ]]` evaluates to false when `"$CONDITION"` is an empty string.

File: .travis.yml [hunk 4]
```diff
@@ -21,13 +21,15 @@ env:
...
      - GCP_PROJECT="fairspace-207108"
-     - SNAPSHOT_CONTEXT="gke_fairspace-207108_europe-west1-b_fairspacecicluster"
-     - SNAPSHOT_CLUSTER_NAME="fairspacecicluster"
-     - SNAPSHOT_CLUSTER_ZONE="europe-west1-b"
-     - RELEASE_CONTEXT="gke_fairspace-207108_europe-west1-b_fairspacecicluster"
-     - RELEASE_CLUSTER_NAME="fairspacecicluster"
-     - RELEASE_CLUSTER_ZONE="europe-west1-b"
      - GIT_TAG_FILES_OVERRIDE="charts/fairspace/Chart.yaml charts/fairspace/values.yaml projects/"
+     - SHOULD_DEPLOY=false
+     # DEPLOYMENT RELATED VARIABLES - REQUIRED IF SHOULD_DEPLOY=1
```
Comment (gijskant) [inline .travis.yml]:
```suggestion
     # DEPLOYMENT RELATED VARIABLES - REQUIRED IF SHOULD_DEPLOY=true
```

**Review by gijskant — APPROVED** [review]

File: .travis.yml (hunk 1 above)
Comment (ewelinajozwik) [reply .travis.yml]: yeah, this is so intuitive... :sweat_smile:

**Pull request merged**